### ISSUE TITLE
extra memory usage in clean_state

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Castro#353]
`clean_state` is a bit wasteful with memory.  In particular, there are 2 interfaces:

```
Castro::clean_state(int is_new, int ng)
```
and
```
Castro::clean_state(int is_new, MultiFab& state_old, int ng) 
```
The latter uses the MF passed in to compute the `frac_change` from the `enforce_min_density()` call.  The former interfaces copies the desired `StateData` to a temporary `MF` and then calls the other `clean_state` interface with this MF.  

It's not clear we always want to do this copy, since sometimes we are calling `clean_state` without regard to the return value (like in `expand_state`).  In these cases, we should create an option that allows us to skip this copy (perhaps by passing two copies of the same thing into `enforce_min_density`.

**Comment by zingale** [comment]
notes from slack:

* the point of the copy was to avoid aliasing in Fortran
* we can perhaps create a separate interface that works on only one state

**Comment by maxpkatz** [comment]
Noting here that the reason enforce_min_density has a return value is that we use to determine if we need a retry, so we're recording whether any zones had a negative density after the hydro update, and how negative they were. The latter is now not really needed because the dynamic retry capability allows us to just try a timestep half as long and hope it works, and the former can be achieved by just checking if state.min(Density) < 0 after the hydro update and storing it in a bool.

**Comment by maxpkatz** [comment]
This will be addressed in #538

**Comment by maxpkatz** [comment]
Resolved in #538
